### ISSUE TITLE
non login user for contact

### ISSUE THREAD

**Issue by gannebamm** [GeoNode/geonode#10342]
**Is your feature request related to a problem? Please describe.**

As a research institute, we need to manage data given by third parties. Since we do not initially create the data, we want to set the originator/author contact to the third party. An example is data from the Federal Agency for Cartography and Geodesy (BKG) (https://gdz.bkg.bund.de/index.php/default/digitale-geodaten.html).

**Describe the solution you'd like**
As an admin, I want to create a `Profile` object which is not allowed to log in.

**Describe alternatives you've considered**
* Use the current implemented `Profile` model for third parties:
https://github.com/GeoNode/geonode/blob/99b0557da5c7db23c72ad39e466b88fe43edf82d/geonode/people/models.py#L64-L65
   These users, in theory, could log in/request a password reset by the email contact info given.
* Create a completely new model for `contacts`. This new model would need to keep in sync with `Profile` and needs new UI elements to give information about the contact. This feels like overhead, seeing that we already have a proper `Profile` implementation which provides these features.

**Additional context**
See this #10290 as a reference for our implementation idea to set multiple contacts to one dataset. After that PR is finished, we will have the following ISO contact roles at hand:

author: party who authored the resource **<< THIRD PARTY ??**
publisher: party who published the resource **<< THIRD PARTY ??**
resource provider: party that supplies the resource **<< THIRD PARTY ??**
originator: party who created the resource  **<< THIRD PARTY ??**
owner: party that owns the resource **<< IN THEORY THIRD PARTY, but this will mix up the user permissions! So keeping it on the resources owner in GeoNode (likely the admin) shall be used as a good enough workaround. See also https://github.com/GeoNode/geonode/issues/10290#issuecomment-1313425520**

**Comment by gannebamm** [comment]
After doing some more digging into the abstract user from Django, maybe this is also ok:
https://docs.djangoproject.com/en/4.1/ref/contrib/auth/#django.contrib.auth.models.User.is_active

Defining a `Profile` as non-active should prevent it from logging in, I think?

**Comment by t-book** [comment]
@gannebamm yes if you remove the active check the user should not be able to login. Does this solve your issue?

![image](https://user-images.githubusercontent.com/20478652/219324927-165ccbba-42bc-408e-9b28-4dfbe605794e.png)

**Comment by t-book** [comment]
(and maybe craete a group for those profiles to quickly identify them)

**Comment by giohappy** [comment]
@gannebamm moreover, those users could be removed from the `contributors` (and `registered-members` if configured) Django Group.

![image](https://user-images.githubusercontent.com/571129/219338722-7b7b295b-6eab-4372-9366-5ff5db4449b3.png)

**Comment by t-book** [comment]
Good point @giohappy Maybe even after every registration with some signal if the email does not end with the company domain...

```
from django.contrib.auth.models import User, Group
from django.dispatch import receiver
from django.db.models.signals import post_save
from django.contrib.auth import get_user_model

@receiver(post_save, sender=get_user_model())
def remove_user_from_all_groups(sender, instance, created, **kwargs):
    if created and not instance.email.endswith("example.com"):
        groups = Group.objects.filter(user=instance)
        for group in groups:
            group.user_set.remove(instance)
```

**Comment by gannebamm** [comment]
I have tested the ideas above. Those will not cover our use-case since you can not set a non active user as a contact. We will discuss options to resolve those requirements also with respect to #11511 and #12124.

**Comment by giohappy** [comment]
Closing as part of a repository cleanup: this issue was last updated before 2025-01-01. Not a judgment on the report itself — feel free to reopen or file a fresh issue against `main` if it's still relevant. Thanks for the report.
